services/gmail_oauth_connection_service.py

```
@@ -0,0 +1,279 @@
+            "refresh_token": self.refresh_token,
+            "grant_type": "refresh_token",
+        }).encode("utf-8")
+
+        request = urllib.request.Request(
+            self.TOKEN_URI,
+            data=data,
+            headers={"Content-Type": "application/x-www-form-urlencoded"},
+        )
+
+        try:
+            with urllib.request.urlopen(request, timeout=30) as response:
```

Ruff (0.14.8)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)